tracer/src/Datadog.Trace/Agent/Api.cs

```diff
@@ -351,7 +357,15 @@ private async Task<bool> SendTracesAsyncImpl(IApiRequest request, bool finalTry,
                 response?.Dispose();
             }
 
-            _log.Debug<int>("Successfully sent {Count} traces to the Datadog Agent.", numberOfTraces);
+            if (response.StatusCode == 429)
+            {
+                var retryAfter = response.GetHeader("Retry-After");
+                _log.Debug<int, string>("Failed to submit {Count} traces. Agent responded with 429 Too Many Requests, retry after {RetryAfter}", numberOfTraces, retryAfter ?? "unspecified");
```

Comment:
Thanks, this looks good. One other thing I just realised though. There's various stats and metrics that we send based on the return value. We could move _some_ of those metric values into this method instead (drop count etc), but not all of it (`TraceKeepRateCalculator`), so we might need a bigger refactor.

I think the easiest approach is probably to switch this method from returning a `bool` to a private `enum`, something like

```csharp
private enum SendTracesResult
{
    Success,
    FailedCanRetry
    FailedDontRetry,
}
```

So that the `SendWithRetry` method knows not to retry the 429 response, but it can also convert the values correctly:

```csharp
return implResult switch
{
    Success => true, // A real success, so AgentWriter increments stats correctly
    _ => false // Could be a 429/404 etc, AgentWriter correctly implements stats
};
```

Reply:
@andrewlock I tried implementing this, but after doing that I'm not entirely sure what exactly else to change metric wise here like you're saying?

Reply:
Great thanks!